src/pages/transactions/_components/DfTx/DfTxSetFutureSwap.tsx

```diff
@@ -10,7 +11,30 @@ interface DfTxSetFutureSwapProps {
 
 export function DfTxSetFutureSwap(props: DfTxSetFutureSwapProps): JSX.Element {
   const from = props.dftx.data.source;
-  const to = props.dftx.data.destination;
+  let to = props.dftx.data.destination;
+  const { connection } = useNetwork();
+
+  /**  
+    This check here is due to 
+    Destination is serialised as 0 in the transaction message in the blockchain if 
+    the source is dToken. When is serialized as 0 token will be shown as DFI instead of DUSD 
+    for dToken -> DUSD swaps 
+  * */
+  if (from.token > 0 && to === 0) {
+    switch (connection) {
+      case "TestNet":
```

Comment:
rearrange the cases by playground > testnet > mainnet

Reply:
updated